example-blog/src/components/Header.astro

```diff
@@ -3,74 +3,69 @@ import HeaderAccount from "../components/HeaderAccount.astro";
 ---
 
 <script>
-  const btnLightElem: HTMLButtonElement | null = document.querySelector(
+  import { atom } from "nanostores";
+
+  const lightThemeButton: HTMLButtonElement | null = document.querySelector(
     "#theme-toggle-light"
   );
-  const btnDarkElem: HTMLButtonElement | null =
+  const darkThemeButton: HTMLButtonElement | null =
     document.querySelector("#theme-toggle-dark");
 
-  function setTheme() {
-    const defaultTheme = window.matchMedia("(prefers-color-scheme: dark)");
-    const theme =
-      localStorage.getItem("theme") ??
-      (defaultTheme.matches ? "dark" : "light");
-    if (btnLightElem && btnDarkElem) {
-      if (theme === "dark") {
-        document.body.classList.remove("theme-light");
-        document.body.classList.add("theme-dark");
-        btnLightElem.style.display = "none";
-        btnDarkElem.style.display = "block";
-      } else {
-        document.body.classList.remove("theme-dark");
-        document.body.classList.add("theme-light");
-        btnLightElem.style.display = "block";
-        btnDarkElem.style.display = "none";
-      }
-    }
-  }
+  const getPreferredTheme = () => {
+    const isDarkThemePreferred = window.matchMedia(
+      "(prefers-color-scheme: dark)"
+    ).matches;
+    return (
+      localStorage.getItem("theme") ?? (isDarkThemePreferred ? "dark" : "light")
+    );
+  };
+
+  const setTheme = (theme: string) => {
+    document.body.className = `theme-${theme}`;
+    if (lightThemeButton)
+      lightThemeButton.style.display = theme === "dark" ? "none" : "block";
+    if (darkThemeButton)
+      darkThemeButton.style.display = theme === "light" ? "none" : "block";
+  };
 
-  function toggleTheme() {
-    const defaultTheme = window.matchMedia("(prefers-color-scheme: dark)");
-    const theme =
-      localStorage.getItem("theme") ??
-      (defaultTheme.matches ? "dark" : "light");
-    if (theme === "dark") {
-      localStorage.setItem("theme", "light");
-      document.body.classList.remove("theme-dark");
-      document.body.classList.add("theme-light");
+  const toggleTheme = () => {
+    const currentTheme = getPreferredTheme();
+    const newTheme = currentTheme === "dark" ? "light" : "dark";
+    localStorage.setItem("theme", newTheme);
+    setTheme(newTheme);
+  };
 
-      if (btnLightElem && btnDarkElem) {
-        btnLightElem.style.display = "block";
-        btnDarkElem.style.display = "none";
-      }
-    } else {
-      localStorage.setItem("theme", "dark");
-      document.body.classList.remove("theme-light");
-      document.body.classList.add("theme-dark");
-      if (btnLightElem && btnDarkElem) {
-        btnLightElem.style.display = "none";
-        btnDarkElem.style.display = "block";
-      }
+  document.addEventListener("DOMContentLoaded", (event) => {
```

Comment:
There's a FUOC currently, consider moving this at the start of the body. You can take a look at how we do it in Pink Design: [appwrite/pink@main/apps/pink/src/layouts/Layout.astro](https://github.com/appwrite/pink/blob/main/apps/pink/src/layouts/Layout.astro)

Reply:
Refactored the theme switcher to it's own component - and added inline script to prevent FOUC